husky_components/datagrid/decorators/table-view.js

```diff
@@ -1424,7 +1424,7 @@ define(function() {
          * @param event {Object} the event object
          */
         allSelectItemChangeHandler: function(event) {
-            if (this.data.total == 0) {
+            if (this.data.embedded.length == 0) {
```

Comment:
No, the problem here is, that the select-all toggle does not work, if there were no server-side list items at the point of last load. If we add/remove elements to the datagrid locally, `this.data.total` will not change and therefore the toggle will not do anything.

Returning a `ListRepresentation` will fix the toggle only after saving, but not before.
I think this should work with pagination, as long we do no have empty pages?